compiler/optimizer/VPHandlers.cpp

```diff
@@ -4187,8 +4187,12 @@ TR::Node *constrainNewArray(OMR::ValuePropagation *vp, TR::Node *node)
       dumpOptDetails(vp->comp(), "size node has no known constraint for newarray %p\n", sizeNode);
       //node->setAllocationCanBeRemoved(false)
       }
-   else
+   else if (sizeConstraint->getLowInt() >= 0 && sizeConstraint->getLowInt() <= maxSize)
+      {
+      // If array size is known to be positive and the number of elements will
```

Comment:
Would you mind saying *non-negative* rather than *positive* here? (and in the comment in `constrainANewArray()`)

Reply:
Thanks for catching that

Reply:
Corrected this in commit f69a2e82d40235de90493904a9342490fdcd9c94